### PR TITLE
More authors

### PR DESCRIPTION
**Pull request by Joozty**
Video with screen reader enabled

https://www.loom.com/share/d1d39f8703814893a15d43c489197cd5

### CODE REVIEW

File: src/modules/search-result/author.jsx [hunk 1]
```diff
@@ -1,28 +1,80 @@
...
-      itemScope=""
-      itemType="https://schema.org/Person"
-      {...htmlProps}
-    >
-      <ElementType href={href} itemprop="name">
-        {children}
-      </ElementType>
-    </li>
+    <ol className={classNames.use(styles.list).join(className)} {...htmlProps}>
+      <Item aria-setsize={authors.length} name={authors[0].name} />
```
Comment (viktor-yakubiv) [inline src/modules/search-result/author.jsx:49]: You did not use the `aria-posinset`, it's recommended by the spec to combine these 2.

File: src/modules/search-result/author.jsx [hunk 2]
```diff
@@ -1,28 +1,80 @@
...
+        className={classNames.use(styles.showMore, isExpanded && styles.hide)}
+      >
+        <Button
+          ref={showMoreRef}
+          onFocus={() => {
+            setIsExpanded(true)
+          }}
+          title="Show more authors"
+        >
+          + {authors.length - 3} MORE
```
Comment (viktor-yakubiv) [inline src/modules/search-result/author.jsx:62]:
```suggestion
          +{authors.length - 3} more
```

Change the case using `text-transform: uppercase` but keep the text always in the correct case. I would call it design's backward compatibility.

If you want to add a space after `+` use some thinner one, a thin or a hair would be good, or up to your eye. Use [Unicode spaces](http://jkorpela.fi/chars/spaces.html) as a reference.

File: src/modules/search-result/author.jsx [hunk 3]
```diff
@@ -1,28 +1,80 @@
...
 
-const Item = ({ children, href, ...htmlProps }) => {
-  const ElementType = href != null ? 'a' : 'span'
+const List = ({ authors, className, ...htmlProps }) => {
+  const showMoreRef = useRef(null)
+  const [isExpanded, setIsExpanded] = useState(false)
+
+  useEffect(() => {
+    if (isExpanded)
+      showMoreRef.current.parentElement.nextElementSibling.firstChild.focus()
```
Comment (viktor-yakubiv) [inline src/modules/search-result/author.jsx:31]: I am afraid this effect will work before the actual click. Let me check how exactly it works... ⏳

**Review by viktor-yakubiv — APPROVED** [review]
I like what I've seen, I like how it's implemented. It's very good 👍

There are some remarks in the comment that are easier to fix. Maybe I will improve this in the future after some checks but this is nice now.

Thank you!

File: src/modules/search-result/author.jsx (hunk 3 above)
Comment (Joozty) [reply src/modules/search-result/author.jsx:31]: there is `useLayoutEffect` which I wanted initially use but then it worked even with useEffect so I went with it.  (`useLayoutEffect` is just client side hook and I didn't want to break the SSR)

File: src/modules/search-result/metadata.css [hunk 4]
```diff
@@ -6,7 +6,7 @@
 .group {
   display: inline-block;
 
-  &:not(:last-child)::after {
+  &:not(:first-child)::before {
```
[the remark this answers is not in the document]
Comment (Joozty) [reply src/modules/search-result/metadata.css:9]: I wanted avoid this empty line:

::after
after
::before
before

File: src/modules/search-result/author.jsx (hunk 1 above)
Comment (Joozty) [reply src/modules/search-result/author.jsx:49]: Isn't it only required when I want to have a different order of items?

File: src/modules/search-result/author.jsx (hunk 3 above)
Comment (viktor-yakubiv) [reply src/modules/search-result/author.jsx:31]: Yes, it does. Once I mouse-down it, the list changes but the click should happen only when the mouse is released.

It's not about the `useEffect` hook, it's the way it's implemented.

Comment (Joozty) [reply src/modules/search-result/author.jsx:31]: Alright but I don't think this is a big deal, is it?

File: src/modules/search-result/metadata.css (hunk 4 above)
[the remark this answers is not in the document]
Comment (viktor-yakubiv) [reply src/modules/search-result/metadata.css:9]: I think you can avoid it converting the list to `inline`. In my opinion, it's the right way.

File: src/modules/search-result/author.jsx (hunk 3 above)
Comment (viktor-yakubiv) [reply src/modules/search-result/author.jsx:31]: It's not 🙂

File: src/modules/search-result/metadata.css (hunk 4 above)
[the remark this answers is not in the document]
Comment (Joozty) [reply src/modules/search-result/metadata.css:9]: Not really it didn't work for me. If you want you can try it. However I will merge this one, feel free to improve it revert it back if you find a solution for it.

**Pull request merged**